### ISSUE TITLE
Steps for generating CF-compliant files

### ISSUE THREAD

**Issue by dnowacki-usgs** [USGS-CMG/usgs-cmg-portal#306]
Note from phone call with @rsignell-usgs on 2019-05-09.

Take a simple St Pete or Santa Cruz dataset. Write a script which:

- Uses xarray to add necessary CF metadata and migrate EPIC time/time2 to CF time.
- Ensures CF-compliance either using [ioos/compliance-checker](https://github.com/ioos/compliance-checker) or the [API](https://github.com/ioos/compliance-checker-web/wiki/API)
- Writes ERDDAP snippet
- Toss it on gamone, add the ERDDAP snippet, and enjoy your newly discoverable SP/SC dataset

**Comment by rsignell-usgs** [comment]
@dnowacki-usgs , I asked @joefutrelle at WHOI if OOI was using [pocean-core](https://github.com/pyoceans/pocean-core) and he said they were migrating toward `xarray`.  OOI's  @cwingard is going most of this work over on [this bitbucket repo]().   It seems pretty hard-wired for OOI ([example here](https://bitbucket.org/ooicgsn/cgsn-processing/src/0c5336f3c359a0783a5a05e71a5e6bfa26131b14/cgsn_processing/process/proc_adcp.py#lines-178:184)), however, so thinking about a more flexible design based on `xarray` with these folks would be worth it, perhaps with input from @kwilcox  and @ocefpaf as well?

**Comment by kwilcox** [comment]
I don't disagree. If you are starting with tabular data then `pocean-core` is your best bet. If you are starting from netCDF files then some manipulation in `xarray` will work better. I'd be interested in enhancements to `pocean-core` to support indexing an `xarray` dataset to match the DSG types and being to export them as netCDF.

**Comment by ocefpaf** [comment]
>  If you are starting with tabular data then `pocean-core` is your best bet.

Yep. You can avoid writing a lot of code by leveraging with `pocean-core`, not to mentioned that any bug fixes would benefit a larger community.
